--- Storage.swift    2023-04-21 12:00:00
+++ Storage.swift    2023-06-12 12:00:00
@@ -69,7 +69,7 @@
   public var useEmulator: Bool {
     get {
       // Firestore uses a different approach to opt in to the emulator.
-      return provider?.storage.useEmulator ?? false
+      return provider != nil ? provider!.storage.useEmulator : false
     }
     set {
       // This empty setter is here to satisfy compiler, because this property should be read only from outside.
@@ -84,7 +84,7 @@
   /// The default Storage bucket's host.
   public var host: String {
     get {
-      return provider?.storage.host ?? "https://firebasestorage.googleapis.com"
+      return provider != nil ? provider!.storage.host : "https://firebasestorage.googleapis.com"
     }
     set {
       // This empty setter is here to satisfy compiler, because this property should be read only from outside.
@@ -287,9 +287,9 @@
 
   /// Provides necessary Firebase dependencies.
   private func createProvider(app: FirebaseApp?) -> StorageProvider {
-    let auth: AuthInterop = AppCheckTokenProviderFactory.createAuthTokenProvider(app: app)
+    let auth: AuthInterop = AppCheckTokenProviderFactory.createAuthTokenProvider(app: app) as AuthInterop
     let appCheck: AppCheckInterop? = AppCheckTokenProviderFactory.createAppCheckTokenProvider(app: app)
-    return StorageProvider(app: app, auth: auth, appCheck: appCheck)
+    return StorageProvider(app: app, auth: auth, appCheck: appCheck as? AppCheckInterop)
   }
 
   public func reference(for url: URL) -> StorageReference {
